src/test/ui/consts/extra-const-ub/detect-extra-ub.rs

```diff
@@ -0,0 +1,43 @@
+// compile-flags: -Zextra-const-ub-checks
```

Comment:
maybe change this to use revisions so that this test doesn't get outdated once we detect some of these cases without this flag?

Reply:
Note sure what you mean? Even if we do, we'll still also want to detect them with the flag, so the test remains correct?

Reply:
this test is explicitly about ub detected by `-Zextra-const-ub-checks`. If all of the UB in this test file also gets detected without this flag, then the flag becomes useless.

My idea is to add a check that without this flag, we have UB that doesn't cause an error.

Reply:
sure, by adding `// [no-flag] check-pass`

Reply:
Oh, I didn't know these revision things also work on the check-pass flags, neat!